Decode morse

RileyManda opened the pull request:
- [x] Added readme
- [x] Project setup and initialization
- [x] Updated folder name
- [x] Linter fixes
-  Create a method to decode the entire message in Morse code
-   Create a method to decode a Morse code character
- Create a method to decode the entire word

RileyManda committed:
Added readme

RileyManda committed:
Project setup and initialization

RileyManda committed:
Updated folder name

RileyManda committed:
Linter fixes

RileyManda committed:
Created a method to decode a Morse code character

RileyManda committed:
Removed white spaces from the string

RileyManda committed:
Added comments

ClarenceAtim committed:
Create a method to decode the entire message in Morse code

RileyManda committed:
Created a method to decode a morse code character

RileyManda committed:
removed additinal method

RileyManda committed:
Linter fixes

V-Blaze commented on decoding_morse.rb:
```diff
@@ -0,0 +1,33 @@
+  end
+
+  decoded_words.join('   ')
+end
```
- You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :smile: :wave: 

- [ ] According to the project requirement, you are meant to run this test to see your decode morse code in action.

```sh
p decode_char('.-')
p decode_word('-- -.--')
p decode_message("-- -.--   -. .- -- .")
p decode_message(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")
```
-  [ ] When I ran these, I got an error in the console, I'll suggest you go through your methods and make sure everything is implemented correctly, and then run the code to make sure the result you get is what's expected. :smile: :small_airplane: :wave: 

- - Screenshot for reference:
![Screenshot from 2023-08-30 11-59-13](https://github.com/RileyManda/morse-code/assets/107632780/73370a66-2903-435d-a29a-dbb7c2bc55cf)

V-Blaze commented on README.md:
```diff
@@ -1 +1,244 @@
-# morse-code
\ No newline at end of file
+<a name="readme-top"></a>
+
+<div align="center">
+
+  <img src="app-logo.png" alt="logo" width="140"  height="auto" />
+  <br/>
```
- [ ] You are meant to remove this Microverse logo as it is a placeholder that came with the template. You can add your logo if you have any or leave out the logo entirely. :small_airplane: :wave: 

- - Screenshot:
![Screenshot from 2023-08-30 12-03-10](https://github.com/RileyManda/morse-code/assets/107632780/fd698154-9466-4d5a-a3f2-8dfec9d20a47)

RileyManda replied:
Noted with Thanks @V-Blaze

RileyManda committed:
bug fix

V-Blaze commented on README.md:
```diff
@@ -1 +1,244 @@
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+[![MIT License](https://img.shields.io/badge/License-MIT-green.svg)](./LICENSE)
+
```
- [ ] The License link you have in your readme leads to a 404 error page. I'll suggest you add a Licence file and correctly link it in your readme. :smile: :wave: 

- - Screenshot:
![Screenshot from 2023-08-30 12-05-45](https://github.com/RileyManda/morse-code/assets/107632780/326d5324-0e3b-4b74-aa36-f4507ea877ce)

RileyManda committed:
Linter fixes

V-Blaze commented on README.md:
```diff
@@ -1 +1,244 @@
+
+<!-- ATTRIBUTION -->
+
+## 👥 Attribution <a name="attribution"></a>
+- This project is based on the Original design by: [Nelson Sakwa
+](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs))
+- Original logo design from: [Svgrepo](https://www.svgrepo.com/)
```
- [ ] This project's design is not from `Nelson Sakwa`, I'll suggest you update your readme with accurate information about the project as a good-looking readme makes your project look professional. :smile: :small_airplane:

RileyManda replied:
Thank you for the review @V-Blaze 
The readme file template was copied from a previous project.
We will make the changes to submit for review.

V-Blaze reviewed (changes requested):
## Hi @RileyManda, @ClarenceAtim  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

![giphy](https://user-images.githubusercontent.com/107632780/229314554-3700de13-2c00-46d5-bc77-46ad09af7467.gif)


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Well-documented code✔
-  Correct Git flow👋 
-  Linters setup correctly ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda committed:
Updated readme

RileyManda committed:
Updated installation instructions

RileyManda commented:
We have completed making the requested changes from the previous reviewer:

-Ensured the following commands are executing successfully on the terminal:
p decode_char('.-')
p decode_word('-- -.--')
p decode_message("-- -.--   -. .- -- .")
p decode_message(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")


<img width="950" alt="Screen Shot 2023-08-30 at 1 28 54 PM" src="https://github.com/RileyManda/morse-code/assets/37291194/19d8c713-2f4b-4633-af27-1c0ad378fff7">

-Updated readme file and remove attributions  and project setup
-Updated readme file and remove logo

RileyManda committed:
Added licence to project

LionRouge1 commented on decoding_morse.rb:
```diff
@@ -0,0 +1,33 @@
+end
+
+# Create a method to decode the entire message in Morse code
+def decode_message(morse_message)
+  words = morse_message.split('   ')
+  decoded_words = words.map do |word|
+    characters = word.chars
+    decoded_characters = characters.map { |char| decode_char(char) }
+    decoded_characters.join
+  end
+
+  decoded_words.join('   ')
+end
```
- [ ] Good job creating the `decode_char` and the `decode_word` separately :clap: However the `decode_message` function is not returning the correct message.
  This `decode_message("-- -.--   -. .- -- .")`  supposed to returns **`"MY NAME"`** but we got this **`"TTUNKNOWNTETT   TEUNKNOWNETUNKNOWNTTUNKNOWNE"`**
  Kindly make sure the correct message is returned by the `decode_message` function.

  **HIT:  We can simply use the `decode_word` function in our `decode_message` function. So the `decode_message` could looks this :point_down:** (Your code doesn't need to be exactly like this)

```ruby
def decode_message(morse_message)
  morse_message.split('   ').map { |w| decode_word(w) }.join(' ')
end
```

LionRouge1 reviewed (changes requested):
Hi @RileyManda && @ClarenceAtim, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![almost](https://media.giphy.com/media/R2m2NzVxQ3pbG/giphy.gif)

## Suggested changes :recycle: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda commented:
<img width="427" alt="Screen Shot 2023-08-30 at 2 45 38 PM" src="https://github.com/RileyManda/morse-code/assets/37291194/9312720a-0ecc-48ef-97d8-3dd5e451d511">

<img width="813" alt="Screen Shot 2023-08-30 at 3 19 18 PM" src="https://github.com/RileyManda/morse-code/assets/37291194/65dd3acb-af16-40f7-9747-67adaf638e68">

RileyManda committed:
Bug fix in decoding message

RileyManda committed:
Removed blank line-linter fix

RileyManda committed:
New line-linter fix

ClarenceAtim committed:
Updated dictionary and project structure

RileyManda committed:
Linrer fixes

joel-kalema reviewed (approved):
Hi @RileyManda 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 

![ ](https://media4.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif)
## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
